backend/src/main/kotlin/org/loculus/backend/utils/MetadataEntry.kt

```diff
@@ -60,40 +67,44 @@ fun revisionEntryStreamAsSequence(metadataInputStream: InputStream): Sequence<Re
         CSVFormat.TDF.builder().setHeader().setSkipHeaderRecord(true).build(),
     )
 
-    if (!csvParser.headerNames.contains(HEADER_TO_CONNECT_METADATA_AND_SEQUENCES)) {
-        throw UnprocessableEntityException(
-            "The revised metadata file does not contain the header '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES'",
+    val headerNames = csvParser.headerNames
+    val submissionIdHeaders = listOf(
+        HEADER_TO_CONNECT_METADATA_AND_SEQUENCES,
+        HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT,
+    ).filter { headerNames.contains(it) }
+
+    when {
+        submissionIdHeaders.isEmpty() -> throw UnprocessableEntityException(
+            "The revised metadata file does not contain either header '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES' or '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT'",
+        )
+        submissionIdHeaders.size > 1 -> throw UnprocessableEntityException(
+            "The revised metadata file contains both '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES' and '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT'. Only one is allowed.",
```

Comment:
I don't quite understand this code, isn't the following clearer?

```suggestion
val containsHeader = csvParser.headerNames.contains(HEADER_TO_CONNECT_METADATA_AND_SEQUENCES)
val containsHeaderBackcompat = csvParser.headerNames.contains(HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT)

if (containsHeader1 && containsHeader2) {
    throw UnprocessableEntityException(
        "The revised metadata file contains both '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES' and '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT'. Only one is allowed.",
    )
}
if (!containsHeader && !containsHeader2) {
    throw UnprocessableEntityException(
        "The revised metadata file does not contain either header '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES' or '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT'"
    )
}
```

Reply:
I refactored to pull out a shared function which should make it more readable in that the intention is clearer